cmd/osbuild-composer/main.go

```diff
@@ -67,5 +68,13 @@ func main() {
 	weldrAPI := weldr.New(rpm, currentArch(), distribution, logger, store)
 
 	go jobAPI.Serve(jobListener)
+
+	// Optionally run RCM API as well as Weldr API
+	if len(listeners) == 3 {
```

Comment:
This is a fine first version, but is there a way to get more information out of systemd about the passed sockets so we don't have to make assumptions based purely on the number of sockets passed?

Reply:
I'll investigate.

Reply:
I think it is only possible to assign a name to all sockets in one socket unit:
https://www.freedesktop.org/software/systemd/man/systemd.socket.html#FileDescriptorName=

Reply:
You can also take a look here:
https://github.com/coreos/go-systemd/blob/master/activation/listeners.go#L42

Reply:
I might have misunderstood the use-case here. Do we want to create a 2nd socket file specifically for RCM? If so, we can distinguish the sockets by the unit in which they were defined. (I think)

Reply:
I think having one unit file per socket would be nice. Then people can disable/enable/mask to their heart's content, and get exactly the API's they need. And also, as you say, we get to distinguish between them.

Reply:
But we can do that in a follow-up, let's get this in as-is first.

Reply:
systemd sets some environmental variables when it starts the service, I found mention of $LISTEN_FDNAMES here - https://www.freedesktop.org/software/systemd/man/sd_listen_fds.html which may be useful. It doesn't seem to be very well documented though.